src/Synapse/Synapse/Models/SynapseAnalyticsRoleClient.cs

```diff
@@ -29,45 +30,51 @@ public SynapseAnalyticsRoleClient(string workspaceName, IAzureContext context)
 
             string suffix = context.Environment.GetEndpoint(AzureEnvironment.ExtendedEndpoint.AzureSynapseAnalyticsEndpointSuffix);
             Uri uri = new Uri("https://" + workspaceName + "." + suffix);
-            _accessControlClient = new AccessControlClient(uri, new AzureSessionCredential(context));
+            _roleAssignmentsClient = new RoleAssignmentsClient(uri, new AzureSessionCredential(context));
+            _roleDefinitionsClient = new RoleDefinitionsClient(uri, new AzureSessionCredential(context));
             _activeDirectoryClient = new ActiveDirectoryClient(context);
         }
 
-        public IReadOnlyList<RoleAssignmentDetails> ListRoleAssignments(string roleDefinitionId = null, string objectId = null, string continuationToken = null)
+        public RoleAssignmentDetailsList ListRoleAssignments(string roleDefinitionId = null, string objectId = null, string continuationToken = null)
         {
-            return _accessControlClient.GetRoleAssignments(roleDefinitionId, objectId, continuationToken).Value;
+            return _roleAssignmentsClient.ListRoleAssignments(roleDefinitionId, objectId, continuationToken).Value;
```

Comment:
>ListRoleAssignments [](start = 42, length = 19)

This API can support "scope" parameter. Do we need to add a corresponding parameter in cmdlet too?